examples/reference/custom_components/JSComponent.md

```diff
@@ -0,0 +1,428 @@
+    });
+    data.watch(() => {
+        btn.innerHTML = `count is ${data.value}`;
+    }, 'value');
+    return btn;
+}
+```
+
+Now create the file **counter_button.css**.
+
+```css
+button {
+    background: #0072B5;
+    color: white;
+    border: none;
+    padding: 10px;
+    border-radius: 4px;
+}
+button:hover {
+    background: #4099da;
+}
+```
+
+Serve the app with `panel serve counter_button.py --autoreload`.
+
+You can now edit the JavaScript or CSS file, and the changes will be automatically reloaded.
+
+- Try changing the `innerHTML` from `count is ${data.value}` to `COUNT IS ${data.value}` and observe the update.
+- Try changing the background color from `#0072B5` to `#008080`.
+
+## Displaying A Single Panel Component
+
+You can display Panel components by defining a `Child` parameter.
+
+Lets start with the simplest example:
+
+```python
+import param
+import panel as pn
+
+from panel.custom import Child, JSComponent
+
+class Example(JSComponent):
+
+    child = Child()
+
+    _esm = """
+    export function render({ children }) {
+      const button = document.createElement("button");
+      button.appendChild(children.child)
+      return button
+    }"""
+
+Example(child=pn.panel("A **Markdown** pane!")).servable()
+```
+
+If you want to allow a certain type of Panel components only you can specify the specific type in the `class_` argument.
+
+```python
+import param
+import panel as pn
+
+from panel.custom import Child, JSComponent
+
+class Example(JSComponent):
+
+    child = Child(class_=pn.pane.Markdown)
+
+    _esm = """
+    export function render({ children }) {
+      const button = document.createElement("button");
+      button.appendChild(children.child)
+      return button
+    }"""
+
+Example(child=pn.panel("A **Markdown** pane!")).servable()
+```
+
+The `class_` argument also supports a tuple of types:
+
+```python
+import param
+import panel as pn
+
+from panel.custom import Child, JSComponent
+
+class Example(JSComponent):
+
+    child = Child(class_=(pn.pane.Markdown, pn.pane.HTML))
+
+    _esm = """
+    export function render({ children }) {
+      const button = document.createElement("button");
+      button.appendChild(children.child)
+      return button
+    }"""
+
+Example(child=pn.panel("A **Markdown** pane!")).servable()
+```
+
+## Displaying a List of Panel Components
+
+You can also display a `List` of `Viewable` `objects` using the `Children` parameter type:
+
+```python
+import param
+import panel as pn
+
+from panel.custom import Children, JSComponent
+
+class Example(JSComponent):
+
+    objects = Children()
+
+    _esm = """
+    export function render({ children }) {
+      const div = document.createElement('div')
+      div.append(...children.objects)
+      return div
+    }"""
+
+
+Example(
+    objects=[pn.panel("A **Markdown** pane!"), pn.widgets.Button(name="Click me!")]
+).servable()
+```
+
+:::note
+You can change the `item_type` to a specific subtype of `Viewable` or a tuple of
```

Comment:
Would it be possible to rename `item_type` to `class_`? Then `Child` and `Children` would be aligned and there would be less to explain.